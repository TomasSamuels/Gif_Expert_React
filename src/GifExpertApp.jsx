import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";
import { useState } from "react";

export const GifExpertApp = () => {
    
    const [categories, setCategories] = useState(['Attack On Titan']);

    const onAddCategory = ( newCategory ) =>{
        const categoriesLowerCase = categories.map( category => category.toLowerCase());
        if( categoriesLowerCase.includes(newCategory.toLowerCase()) ) return;
        setCategories([ newCategory, ...categories]);
    };

    return (
        <>
        <h1>GifExpertApp</h1>

        <AddCategory
           onNewCategory={ onAddCategory }
        />

        {
            categories.map( category => (
                <GifGrid
                    key={ category }
                    category={ category }
                />
            ))
        }

        </>
    )
}
