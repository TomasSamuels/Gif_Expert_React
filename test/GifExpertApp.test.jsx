import { fireEvent, render, screen } from "@testing-library/react";
import { GifExpertApp } from "../src/GifExpertApp";

describe('GifExpertApp Test', () => { 

    test('should match with snapshot', () => { 
        const { container } = render( <GifExpertApp/> );
        expect( container ).toMatchSnapshot();
    });

    test('should have text "GifExpertApp"', () => { 
        render( <GifExpertApp/> );
        expect( screen.getByText('GifExpertApp') );
    });

    test('should add category', () => { 
        render( <GifExpertApp/> );
        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');
        fireEvent.input(input, { target: { value: 'Naruto' } });
        fireEvent.submit(form);
        expect( screen.getAllByLabelText('category-title').length ).toBe(2);
    });

    test('should not add a category if already exists', () => { 
        render( <GifExpertApp/> );
        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');

        fireEvent.input(input, { target: { value: 'Naruto' } });
        fireEvent.submit(form);

        fireEvent.input(input, { target: { value: 'Naruto' } });
        fireEvent.submit(form);

        expect( screen.getAllByLabelText('category-title').length ).toBe(2);
    });

 });