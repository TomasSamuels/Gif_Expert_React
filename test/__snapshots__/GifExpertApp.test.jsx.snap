// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp Test should match with snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Search gifs"
      type="text"
      value=""
    />
  </form>
  <h3
    aria-label="category-title"
  >
    Attack On Titan
  </h3>
  <h2>
    Loading...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
